File: ruby/expreg_args/expreg.rb
# coding: utf-8

<<DOC
 * Verifica si la cadena word cumple la expresión regular
 *
 * @param word   cadena a evaluar     
 *
 * @return true si cumple, false en otro caso
DOC
def match(word)
  return word.match?(/^[A-Z]([a-z]|ñ){1,5}$/)
end

if ARGV.length != 1
  puts "Recibe un solo argumento\nDebe ser de la forma [A-Z]([a-z]|ñ){1,5}"
else
  if match(ARGV[0])
    puts "Felicidades la cadena es valida!"
  else
    puts "La cadena no es correcta"
  end
end
puts "Cerrando programa ..."

File: go/expreg_args/expreg.go
package main

import(
   "os"
   "fmt"
   "regexp"
)

/*
* Verifica si la cadena str cumple con la expresión regular [A-Z]([a-z]|ñ){1,5}
*
* @param str : string   cadena a evaluar
*
* @return true si cumple, false en otro caso
*/
func match(str string) bool{
  var expreg = regexp.MustCompile("[A-Z]([a-z]|ñ){1,5}") // compila la expresión regular
  return expreg.MatchString(str)  // evalua la cadena con la expresión regular
}

func main(){
  if len(os.Args) != 2 {
    fmt.Println("Debe ser un solo argumento\nDebe ser de la forma [A-Z]([a-z]|ñ){1,5}")
  } else { 
    if match(os.Args[1]) {
      fmt.Println("Felicidades la cadena es valida!")
    } else {
      fmt.Println("Cadena no es correcta")
    }
  }
  fmt.Println("Cerrando programa ...")
}